ddtrace/tracer/spancontext.go

```diff
@@ -304,6 +304,25 @@ func (c *SpanContext) baggageItem(key string) string {
 // finish marks this span as finished in the trace.
 func (c *SpanContext) finish() { c.trace.finishedOne(c.span) }
 
+// safeDebugString returns a safe string representation of the SpanContext for debug logging.
+// It excludes potentially sensitive data like baggage contents while preserving useful debugging information.
+func (c *SpanContext) safeDebugString() string {
+	if c == nil {
+		return "<nil>"
+	}
+
+	hasBaggage := atomic.LoadUint32(&c.hasBaggage) != 0
+	var baggageCount int
+	if hasBaggage {
+		c.mu.RLock()
+		baggageCount = len(c.baggage)
+		c.mu.RUnlock()
+	}
+
+	return fmt.Sprintf("SpanContext{traceID=%s, spanID=%d, hasBaggage=%t, baggageCount=%d, origin=%q, updated=%t, isRemote=%t, baggageOnly=%t}",
```

Comment:
Is there a reason spanLinks is missing here?

Reply:
I wanted to reduce the amount of data we log. If you think it is critical for debugging. I would be happy to add it.